codec/proto_codec.go

```diff
@@ -40,16 +39,9 @@ func (pc *ProtoCodec) MarshalBinaryLengthPrefixed(o ProtoMarshaler) ([]byte, err
 		return nil, err
 	}
 
-	buf := new(bytes.Buffer)
-	if err := encodeUvarint(buf, uint64(o.Size())); err != nil {
-		return nil, err
-	}
-
-	if _, err := buf.Write(bz); err != nil {
-		return nil, err
-	}
-
-	return buf.Bytes(), nil
+	var sizeBuf [10]byte
```

Comment:
```suggestion
	var sizeBuf [binary.MaxVarintLen64]byte
```
nit.

awesome find!!

Reply:
Thank you for the review @marbar3778! Addressed and please take a look.